File: packages/sdk-kotlin/src/main/kotlin/com/flick/sdk/model/FlagConfig.kt
package com.flick.sdk.model

import kotlinx.serialization.Serializable

@Serializable
data class GroupRule(
    val attribute: String,
    val operator: String,
    val value: kotlinx.serialization.json.JsonElement,
)

@Serializable
data class FlagGroup(
    val id: String,
    val rules: List<GroupRule>,
)

@Serializable
data class FlagConfig(
    val key: String,
    val gate_type: String,
    val enabled: Boolean,
    val gate_config: Map<String, kotlinx.serialization.json.JsonElement> = emptyMap(),
    val groups: List<FlagGroup> = emptyList(),
)

@Serializable
data class FullFlagConfig(
    val environment: String,
    val flags: List<FlagConfig>,
    val version: String,
)

@Serializable
data class ApiResponse<T>(
    val data: T,
)

File: packages/sdk-ruby/lib/flick/models/flag_config.rb
# frozen_string_literal: true

module Flick
  GroupRule = Struct.new(:attribute, :operator, :value, keyword_init: true) do
    def self.from_hash(hash)
      new(
        attribute: hash["attribute"],
        operator: hash["operator"],
        value: hash["value"]
      )
    end
  end

  FlagGroup = Struct.new(:id, :rules, keyword_init: true) do
    def self.from_hash(hash)
      new(
        id: hash["id"],
        rules: (hash["rules"] || []).map { |r| GroupRule.from_hash(r) }
      )
    end
  end

  FlagConfig = Struct.new(:key, :gate_type, :enabled, :gate_config, :groups, keyword_init: true) do
    def self.from_hash(hash)
      new(
        key: hash["key"],
        gate_type: hash["gate_type"],
        enabled: hash["enabled"],
        gate_config: hash["gate_config"] || {},
        groups: (hash["groups"] || []).map { |g| FlagGroup.from_hash(g) }
      )
    end
  end
end
